hack/check-subsystem-registry.sh

```
@@ -0,0 +1,209 @@
+  ' "$allowlist_file" | sort -u >"$allowlist_types_file"
+}
+
+resolve_merge_base_range() {
+  if git rev-parse --verify "${base_ref}^{commit}" >/dev/null 2>&1; then
+    if merge_base=$(git merge-base HEAD "${base_ref}" 2>/dev/null); then
+      printf '%s...HEAD' "$merge_base"
+      return
+    fi
+  fi
+
+  if git rev-parse --verify main^{commit} >/dev/null 2>&1; then
```

Shellcheck (0.11.0)

[warning] This { is literal. Check expression (missing ;/\n?) or quote it.

(SC1083)